hs_core/views/resource_rest_api.py

```diff
@@ -845,15 +851,20 @@ def get(self, request, pk, filename):
         redirect_url = f.url.replace('django_irods/download/', 'django_irods/rest_download/')
         return HttpResponseRedirect(redirect_url)
 
-    def post(self, request, pk):
+    def post(self, request, pk, filename):
         """
         Add a file to a resource.
         :param request:
         :param pk: Primary key of the resource (i.e. resource short ID)
+        :param filename: the path to the containing folder in the folder hierarchy
         :return:
+
+        Leaving out filename results in calling a different class function
+        that stores in the root directory
```

Comment:
It's a totally different function... it's not the same function.... will clarify comment.